hypothesis-python/src/hypothesis/strategies/_internal/types.py

```diff
@@ -797,6 +931,18 @@ def resolve_Callable(thing):
     # use of keyword arguments and we'd rather not force positional-only.
     if not thing.__args__:  # pragma: no cover  # varies by minor version
         return st.functions()
+
+    # Concatenate and ParamSpec can never be registered or resolved at Callable arguments
+    if isinstance(thing.__args__[0], Iterable):
+        for stuff in thing.__args__[0]:
+            if stuff in ConcatenateTypes + ParamSpecTypes:
+                raise InvalidArgument(f"{stuff} cannot be arguments in Callables.")
+    # TypeGuard can never be returned by Callable types
+    if thing.__args__[-1] in TypeGuardTypes:
```

Comment:
@Zac-HD This is not as simple cause if we test we have Callable[[],TypeGuard[int]] (TypeGuard cannot be used without arg. And the check will be False. Is that a way to extract the fact that  TypeGuard[int] is a "member" of TypeGuard? Right now I have `type(TypeGuard[int]) = <class 'typing._GenericAlias'>`

Reply:
Yeah, we have a "get origin" helper somewhere; searching for `__origin__` should find it pretty quickly.